src/test/java/org/aleksey/springnew/controller/TodoControllerIT.java

```diff
@@ -0,0 +1,196 @@
+        todoResponseDto.setId(id);
+        todoResponseDto.setTitle(todoCreateDto.getTitle());
+        todoResponseDto.setDescription(todoCreateDto.getDescription());
+        todoResponseDto.setPriority(todoCreateDto.getPriority());
+        todoResponseDto.setStatus(StatusType.IN_PROGRESS.toString());
+        todoResponseDto.setDueDate(localDateTime);
+        todoResponseDto.setUpdatedAt(null);
+        todoResponseDto.setCreatedAt(null);
+
+        String jsonRequest = objectMapper.writeValueAsString(todoCreateDto);
+
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.put("/todos/1")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        TodoResponseDto actual = objectMapper.readValue(result.getResponse().getContentAsString(), TodoResponseDto.class);
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        Assertions.assertEquals(todoResponseDto.getId(), actual.getId());
+        Assertions.assertEquals(todoResponseDto.getTitle(), actual.getTitle());
+        Assertions.assertEquals(todoResponseDto.getDescription(), actual.getDescription());
+        Assertions.assertEquals(todoResponseDto.getDueDate(), actual.getDueDate());
+        Assertions.assertEquals(todoResponseDto.getPriority(), actual.getPriority());
+        Assertions.assertEquals(todoResponseDto.getStatus(), actual.getStatus());
+    }
+
+    @Test
+    @DisplayName("Add update task")
+    void getHistoryTodo_ValidRequest_Success() throws Exception {
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.get("/todos/1/history")
+                        .content("")
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        // Given
+        List<TodoHistoryResponseDto> expected = new ArrayList<>();
+        TodoHistoryResponseDto todoHistoryResponseFirstDto = new TodoHistoryResponseDto();
+        todoHistoryResponseFirstDto.setId(1L);
+        todoHistoryResponseFirstDto.setTodoId(1L);
+        todoHistoryResponseFirstDto.setOldState("old state first");
+        todoHistoryResponseFirstDto.setNewState("new state first");
+        expected.add(todoHistoryResponseFirstDto);
+
+        TodoHistoryResponseDto todoHistoryResponseSecondDto = new TodoHistoryResponseDto();
+        todoHistoryResponseSecondDto.setId(2L);
+        todoHistoryResponseSecondDto.setTodoId(1L);
+        todoHistoryResponseSecondDto.setOldState("old state second");
+        todoHistoryResponseSecondDto.setNewState("new state second");
+        expected.add(todoHistoryResponseSecondDto);
+
+        // Then
+        TodoHistoryResponseDto[] actual = objectMapper.readValue(result.getResponse().getContentAsByteArray(), TodoHistoryResponseDto[].class);
```

Comment:
Could be converted to List directly
```suggestion
List<TodoHistoryResponseDto> actual = objectMapper.readValue(
    result.getResponse().getContentAsByteArray(),
    new TypeReference<List<TodoHistoryResponseDto>>() {}
);
```